test/src/module/command/backupTest.c

```diff
@@ -2270,6 +2299,60 @@ testRun(void)
         TEST_RESULT_VOID(lockRelease(true), "release backup lock");
 
         TEST_RESULT_LOG("P00 DETAIL: match file from prior backup host:" TEST_PATH "/test (0B, 100.00%)");
+
+        // -------------------------------------------------------------------------------------------------------------------------
+        TEST_TITLE("report host/100% progress on truncate result");
+
+        // Create job that skips file
+        job = protocolParallelJobNew(VARSTRDEF("pg_data/test1"), protocolCommandNew(strIdFromZ("x")));
+
+        PackWrite *const resultPack1 = protocolPackNew();
+        const BackupFileResult fileResult = {
+            .manifestFile = STRDEF("pg_data/test1"),
+            .backupCopyResult = backupCopyResultTruncate,
+            .copyChecksum = (Buffer *)HASH_TYPE_SHA1_ZERO_BUF,
+        };
+
+        pckWriteStrP(resultPack1, fileResult.manifestFile);
+        pckWriteU32P(resultPack1, fileResult.backupCopyResult);
+        pckWriteU64P(resultPack1, fileResult.copySize);
+        pckWriteU64P(resultPack1, fileResult.bundleOffset);
+        pckWriteU64P(resultPack1, fileResult.blockIncrMapSize);
+        pckWriteU64P(resultPack1, fileResult.repoSize);
+        pckWriteBinP(resultPack1, fileResult.copyChecksum);
+        pckWriteBinP(resultPack1, fileResult.repoChecksum);
+        pckWritePackP(resultPack1, fileResult.pageChecksumResult);
+
+        pckWriteEndP(resultPack1);
+
+        protocolParallelJobResultSet(job, pckReadNew(pckWriteResult(resultPack1)));
+
+        // Create manifest with file
+        OBJ_NEW_BASE_BEGIN(Manifest, .childQty = MEM_CONTEXT_QTY_MAX)
+        {
+            manifest = manifestNewInternal();
+            HRN_MANIFEST_FILE_ADD(manifest, .name = "pg_data/test1", .size = 10);
+        }
+        OBJ_NEW_END();
+
+        sizeProgress = 0;
+        currentPercentComplete = 4567;
+
+        lockInit(TEST_PATH_STR, cfgOptionStr(cfgOptExecId), cfgOptionStr(cfgOptStanza), lockTypeBackup);
+        TEST_RESULT_VOID(lockAcquireP(), "acquire backup lock");
+
+        TEST_RESULT_VOID(
+            backupJobResult(
+                manifest, STRDEF("host"), storageTest, strLstNew(), job, true, 0, &sizeProgress, &currentPercentComplete),
+            "log truncate result");
+        TEST_RESULT_VOID(lockRelease(true), "release backup lock");
+
+        ManifestFile file = manifestFileFind(manifest, STRDEF("pg_data/test1"));
+        TEST_RESULT_UINT(file.bundleId, 0, "check bundle");
+        TEST_RESULT_UINT(file.size, 0, "check size");
+        TEST_RESULT_BOOL(bufEq(HASH_TYPE_SHA1_ZERO_BUF, BUF(file.checksumSha1, HASH_TYPE_SHA1_SIZE)), true, "checksum eq");
+
+        TEST_RESULT_LOG("P00 DETAIL: backup file host:" TEST_PATH "/test1 (10B->0B, 100.00%)");
     }
```

Comment:
These tests are more indirect than i would like, and we are trying to get rid of the tests directly against backupFile().

I'm working on a test harness to inject changes during the backup, but for now the way we do this is to put these types of tests at 3082 since a different file on the standby will appear to have changed after the backup started. See examples starting at 3121.

Reply:
But that's not going to work because this test does not do bundling. I'll need to think about it.

Reply:
I have committed the new backup test harness, which allows us to modify files during the backup. I've added a test to show the current behavior (337da35a), you can update the test as needed for your fix. Don't forget to put back the ASSERT() that I removed.

Be sure to merge all the commits from main as I have migrated all the backupFile() tests to use the new harness.

Reply:
> Be sure to merge all the commits from main as I have migrated all the backupFile() tests to use the new harness.

Do I need to make a pull request to the main branch or pgbackrest:integration?

Reply:
You don't need to create a new pull request -- just merge the commits from pgbackrest:main into this branch so you get the new test harness.

Reply:
done

Reply:
Yes, this is much better.